src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -127,31 +127,39 @@ protected void tick(Block b) {
                     if (!menu.fits(bucket.get(), getOutputSlots())) {
                         return;
                     }
-
+                    
+                    Block consume = findFluid(fluid);
+                    if (consume == null) return;
+              
                     removeCharge(b.getLocation(), ENERGY_CONSUMPTION);
                     menu.consumeItem(slot);
                     menu.pushItem(bucket.get().clone(), getOutputSlots());
-                    consumeFluid(fluid);
-
+                    consume.setType(Material.AIR);
                     return;
                 }
             }
         }
     }
 
-    private void consumeFluid(Block fluid) {
-        if (fluid.isLiquid()) {
-            if (fluid.getType() == Material.WATER || fluid.getType() == Material.BUBBLE_COLUMN) {
-                // With water we can be sure to find an infinite source whenever we go
-                // further than 2 blocks, so we can just remove the water here and save
-                // outselves all of that computing...
-                fluid.setType(Material.AIR);
-            }
-            else {
-                List<Block> list = Vein.find(fluid, RANGE, block -> isLiquid(block) && block.getType() == fluid.getType());
-                list.get(list.size() - 1).setType(Material.AIR);
+    private Block findFluid(Block fluid) {
+        if (fluid.getType() == Material.WATER || fluid.getType() == Material.BUBBLE_COLUMN) {
+            // With water we can be sure to find an infinite source whenever we go
+            // further than 2 blocks, so we can just remove the water here and save
+            // ourselves all of that computing...
+            return fluid;
+        }
```

Comment:
We should probably also do an `isSource` check for fluid here now that I think about it...
Otherwise people will have infinite water sources, even if the water source isn't infinite.
Simply checking whether the block below (fluid) is a source block should be fine though.